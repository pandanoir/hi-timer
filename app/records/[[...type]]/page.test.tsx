/**
 * @jest-environment jsdom
 */
import 'whatwg-fetch';
import RecordPage from './page';
import { UserProvider } from '@auth0/nextjs-auth0/client';
import { SWRConfig, unstable_serialize } from 'swr';
import { render } from '../../../__tests__/render';
jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

describe('RecordPage', () => {
  test('snapshot', async () => {
    const { findByRole, asFragment } = render(
      <UserProvider
        user={{
          nickname: 'user',
          name: 'user@example.com',
          picture: 'https://example.com/profile.png',
          updated_at: '2023-05-31T11:16:50.834Z',
          email: 'user@example.com',
          email_verified: true,
          sub: 'auth0|000000000000000000000000',
          sid: '000000000000000000000000-0000000',
        }}
      >
        <RecordPage params={{ type: undefined }} />
      </UserProvider>
    );
    await findByRole('combobox');
    expect(asFragment()).toMatchSnapshot();
  });
  test('snapshot with prefetch data', async () => {
    const { findByRole, asFragment } = render(
      <SWRConfig
        value={{
          fallback: {
            [`$inf$${unstable_serialize({
              url: '/api/record/read',
              query: { event: '3x3x3', limit: '100' },
            })}`]: [
              {
                data: [
                  {
                    id: '59804293-21f6-4fde-a899-46faba7d99ca',
                    createdAt: '2023-06-07T14:35:54.224Z',
                    time: 13715,
                    penalty: false,
                    dnf: false,
                    scramble:
                      "L  B2 R' D2 U2 B2 R' U2 B2 U2 B  U  R2 D' F' L' U' R2 B  ",
                    userId: 'auth0|000000000000000000000000',
                    event: '3x3x3',
                  },
                  {
                    id: 'dcd7011f-b77d-40cf-9571-b90c51fa90bd',
                    createdAt: '2023-06-07T14:35:23.760Z',
                    time: 15605,
                    penalty: false,
                    dnf: false,
                    scramble:
                      "L  D2 B2 L2 B2 U2 L  F2 R2 U2 F2 R  U' R  F' D' B2 D2 L  B2 ",
                    userId: 'auth0|000000000000000000000000',
                    event: '3x3x3',
                  },
                  {
                    id: '00bdd111-9882-4759-923d-0fe966b8d6fa',
                    createdAt: '2023-06-07T14:34:43.324Z',
                    time: 11508,
                    penalty: false,
                    dnf: false,
                    scramble:
                      "B' U2 B' U2 B' L2 U2 L2 F' L2 U2 F2 R' D  B' L' R2 D' L  D2 F  ",
                    userId: 'auth0|000000000000000000000000',
                    event: '3x3x3',
                  },
                ],
                hasNextPage: true,
              },
            ],
          },
        }}
      >
        <UserProvider
          user={{
            nickname: 'user',
            name: 'user@example.com',
            picture: 'https://example.com/profile.png',
            updated_at: '2023-05-31T11:16:50.834Z',
            email: 'user@example.com',
            email_verified: true,
            sub: 'auth0|000000000000000000000000',
            sid: '000000000000000000000000-0000000',
          }}
        >
          <RecordPage params={{ type: undefined }} />
        </UserProvider>
      </SWRConfig>
    );
    await findByRole('combobox');
    expect(asFragment()).toMatchSnapshot();
  });
});
