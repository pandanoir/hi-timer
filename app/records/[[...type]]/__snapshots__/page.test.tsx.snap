// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordPage snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1ey6aov"
  >
    <div
      class="chakra-select__wrapper css-dsdwpa"
    >
      <select
        class="chakra-select css-1icl0uu"
        id="event"
      >
        <option
          value="3x3x3"
        >
          3x3x3
        </option>
        <option
          value="2x2x2"
        >
          2x2x2
        </option>
        <option
          value="4x4x4"
        >
          4x4x4
        </option>
        <option
          value="5x5x5"
        >
          5x5x5
        </option>
        <option
          value="6x6x6"
        >
          6x6x6
        </option>
        <option
          value="7x7x7"
        >
          7x7x7
        </option>
      </select>
      <div
        class="chakra-select__icon-wrapper css-iohxn1"
      >
        <svg
          aria-hidden="true"
          class="chakra-select__icon"
          focusable="false"
          role="presentation"
          style="width: 1em; height: 1em; color: currentColor;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="chakra-tabs css-m2m7l3"
    >
      <div
        aria-orientation="horizontal"
        class="chakra-tabs__tablist css-1xhq01z"
        role="tablist"
      >
        <button
          aria-controls="tabs-:r0:--tabpanel-0"
          aria-selected="true"
          class="chakra-tabs__tab css-52dxnr"
          data-index="0"
          id="tabs-:r0:--tab-0"
          role="tab"
          tabindex="0"
          type="button"
        >
          table
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-1"
          aria-selected="false"
          class="chakra-tabs__tab css-52dxnr"
          data-index="1"
          id="tabs-:r0:--tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          graph
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-2"
          aria-selected="false"
          class="chakra-tabs__tab css-52dxnr"
          data-index="2"
          id="tabs-:r0:--tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          daily average
        </button>
      </div>
      <div
        class="chakra-tabs__tab-panels css-8atqhb"
      >
        <div
          aria-labelledby="tabs-:r0:--tab-0"
          class="chakra-tabs__tab-panel css-a5mhaz"
          id="tabs-:r0:--tabpanel-0"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="css-gmuwbf"
          >
            <div
              class="chakra-spinner css-jmuyva"
            >
              <span
                class="css-8b45rq"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          aria-labelledby="tabs-:r0:--tab-1"
          class="chakra-tabs__tab-panel css-a5mhaz"
          hidden=""
          id="tabs-:r0:--tabpanel-1"
          role="tabpanel"
          tabindex="0"
        />
        <div
          aria-labelledby="tabs-:r0:--tab-2"
          class="chakra-tabs__tab-panel css-a5mhaz"
          hidden=""
          id="tabs-:r0:--tabpanel-2"
          role="tabpanel"
          tabindex="0"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`RecordPage snapshot with prefetch data 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-ezbzms"
  >
    <div
      class="chakra-select__wrapper css-dsdwpa"
    >
      <select
        class="chakra-select css-9zgf1d"
        id="event"
      >
        <option
          value="3x3x3"
        >
          3x3x3
        </option>
        <option
          value="2x2x2"
        >
          2x2x2
        </option>
        <option
          value="4x4x4"
        >
          4x4x4
        </option>
        <option
          value="5x5x5"
        >
          5x5x5
        </option>
        <option
          value="6x6x6"
        >
          6x6x6
        </option>
        <option
          value="7x7x7"
        >
          7x7x7
        </option>
      </select>
      <div
        class="chakra-select__icon-wrapper css-iohxn1"
      >
        <svg
          aria-hidden="true"
          class="chakra-select__icon"
          focusable="false"
          role="presentation"
          style="width: 1em; height: 1em; color: currentColor;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="chakra-tabs css-14us3sx"
    >
      <div
        aria-orientation="horizontal"
        class="chakra-tabs__tablist css-1xhq01z"
        role="tablist"
      >
        <button
          aria-controls="tabs-:r0:--tabpanel-0"
          aria-disabled="false"
          aria-selected="true"
          class="chakra-tabs__tab css-52dxnr"
          data-index="0"
          id="tabs-:r0:--tab-0"
          role="tab"
          tabindex="0"
          type="button"
        >
          graph
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-1"
          aria-disabled="false"
          aria-selected="false"
          class="chakra-tabs__tab css-52dxnr"
          data-index="1"
          id="tabs-:r0:--tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          daily average
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-2"
          aria-disabled="false"
          aria-selected="false"
          class="chakra-tabs__tab css-52dxnr"
          data-index="2"
          id="tabs-:r0:--tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          table
        </button>
      </div>
      <div
        class="chakra-tabs__tab-panels css-8atqhb"
      >
        <div
          aria-labelledby="tabs-:r0:--tab-0"
          class="chakra-tabs__tab-panel css-a5mhaz"
          id="tabs-:r0:--tabpanel-0"
          role="tabpanel"
          tabindex="0"
        >
          <label
            class="chakra-form__label css-xmbiaz"
          >
            number of records:
            <div
              class="chakra-select__wrapper css-dsdwpa"
            >
              <select
                class="chakra-select css-9zgf1d"
              >
                <option
                  value="100"
                >
                  100
                </option>
                <option
                  value="200"
                >
                  200
                </option>
                <option
                  value="500"
                >
                  500
                </option>
                <option
                  value="1000"
                >
                  1000
                </option>
                <option
                  value="2500"
                >
                  2500
                </option>
              </select>
              <div
                class="chakra-select__icon-wrapper css-iohxn1"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-select__icon"
                  focusable="false"
                  role="presentation"
                  style="width: 1em; height: 1em; color: currentColor;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </label>
          <div
            class="chakra-card css-pxcwhw"
          >
            <div
              class="chakra-spinner css-12395wm"
            >
              <span
                class="css-8b45rq"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          aria-labelledby="tabs-:r0:--tab-1"
          class="chakra-tabs__tab-panel css-a5mhaz"
          hidden=""
          id="tabs-:r0:--tabpanel-1"
          role="tabpanel"
          tabindex="0"
        />
        <div
          aria-labelledby="tabs-:r0:--tab-2"
          class="chakra-tabs__tab-panel css-a5mhaz"
          hidden=""
          id="tabs-:r0:--tabpanel-2"
          role="tabpanel"
          tabindex="0"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
